zebra-consensus/src/checkpoint.rs

```diff
@@ -905,9 +905,12 @@ where
         });
 
         async move {
-            commit_finalized_block
-                .await
-                .expect("commit_finalized_block should not panic")
+            let fut = commit_finalized_block.await;
+            if zebra_chain::shutdown::is_shutting_down() {
+                Err(VerifyCheckpointError::Finished)
```

Comment:
I am unsure if this will be the best to return.

Reply:
I don't think that `Finished` is the same error as `shutdown failure`, but they are easy to confuse.

Let's change the description of `Finished` to `checkpoint request after the final checkpoint has been verified`. That will help avoid confusion in future.

We also only need to return an error if the spawned task actually failed, so we should use `map_err`.

Finally, we might want to wrap the error with a note that Zebra is shutting down. I'm not sure if the code I've written will work, @yaahc might be able to give you a quick fix that does. But if it takes more than 5 minutes, let's just skip the `wrap_err`.

```suggestion
                result.map_err(wrap_err("Zebra is shutting down")).map_err(VerifyCheckpointError::CommitFinalized)
```

Reply:
This wont be easy to do from within `zebra-consensus` because of the `tower::Buffer` inbetween. If we were working directly with `eyre::Report`s here we could just shove a `.note` on the error to get a new section, but until I make the error handling refactors we should probably not bother worrying about notes or suggestions in errors coming from our library APIs

Reply:
Ok, so we still have to fix up the `map_err` and the `Finished` description here.

Reply:
what about something as https://github.com/ZcashFoundation/zebra/pull/1637/commits/1326ceb4189b9d2475b399c53616f2ddadbc8c34 ?

Reply:
Looks good